File: app/src/main/java/mydebts/android/app/feature/addevent/EventsAdapter.java
package mydebts.android.app.feature.addevent;

import android.support.v7.widget.RecyclerView;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import mydebts.android.app.R;

public class EventsAdapter extends RecyclerView.Adapter<EventsAdapter.EventViewHolder> {

    int itemCount = 1;

    @Override
    public EventViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        EventViewHolder holder = EventViewHolder.create(parent);

        TextWatcher emptyTextWatcher = new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if (TextUtils.isEmpty(holder.name.getText())
                        && TextUtils.isEmpty(holder.price.getText())
                        && itemCount > 1) {
                    notifyItemRemoved(holder.getAdapterPosition());
                    itemCount--;
                } else if (!TextUtils.isEmpty(holder.name.getText())
                        && !TextUtils.isEmpty(holder.price.getText())
                        && holder.getAdapterPosition() == itemCount - 1) {
                    notifyItemInserted(itemCount++);
                }
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        };

        holder.name.addTextChangedListener(emptyTextWatcher);
        holder.price.addTextChangedListener(emptyTextWatcher);
        holder.price.addTextChangedListener(new MoneyTextWatcher());

        return holder;
    }

    @Override
    public void onBindViewHolder(EventViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return itemCount;
    }

    static class EventViewHolder extends RecyclerView.ViewHolder {

        final EditText name;
        final EditText price;

        EventViewHolder(View itemView) {
            super(itemView);
            name = (EditText) itemView.findViewById(R.id.name);
            price = (EditText) itemView.findViewById(R.id.price);
        }

        static EventViewHolder create(ViewGroup parent) {
            return new EventViewHolder(LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.item_event, parent, false));
        }
    }
}

File: app/src/main/java/mydebts/android/app/feature/person/EventsAdapter.kt
package mydebts.android.app.feature.person

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import mydebts.android.app.R
import mydebts.android.app.data.model.Participant
import mydebts.android.app.extention.setCurrencyText
import mydebts.android.app.extention.toEventDateString

class EventsAdapter : RecyclerView.Adapter<EventsAdapter.ViewHolder>() {
    private var _participants: List<Participant>? = null
    var participants: List<Participant>?
        get() = _participants
        set(value) {
            _participants = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        return ViewHolder.create(parent)
    }

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        val participant: Participant? = _participants?.get(position)
        holder?.eventName?.text = participant?.event?.date?.toEventDateString()
        holder?.price?.setCurrencyText(participant?.debt!!)
    }

    override fun getItemCount(): Int {
        return _participants?.size ?: 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val eventName: TextView = itemView.findViewById(R.id.text1)
        val price: TextView = itemView.findViewById(R.id.text2)

        companion object {
            fun create(parent: ViewGroup?) : ViewHolder {
                return ViewHolder(LayoutInflater.from(parent?.context)
                        .inflate(R.layout.item_two_spans, parent, false))
            }
        }
    }
}